addons/hr_expense/static/src/views/kanban.js

```diff
@@ -0,0 +1,36 @@
+/** @odoo-module */
+
+import { registry } from '@web/core/registry';
+import { patch } from '@web/core/utils/patch';
+
+import { ExpenseDashboardMixin } from '../mixins/expense_dashboard';
+import { ExpenseMobileQRCode } from '../mixins/qrcode';
+import { ExpenseDocumentUpload } from '../mixins/document_upload';
+
+import { kanbanView } from '@web/views/kanban/kanban_view';
+import { KanbanController } from '@web/views/kanban/kanban_controller';
+import { KanbanRenderer } from '@web/views/kanban/kanban_renderer';
+
+export class ExpenseKanbanController extends KanbanController {}
+patch(ExpenseKanbanController.prototype, 'expense_kanban_controller_upload', ExpenseDocumentUpload);
+
+export class ExpenseKanbanRenderer extends KanbanRenderer {}
+patch(ExpenseKanbanRenderer.prototype, 'expense_kanban_renderer_qrcode', ExpenseMobileQRCode);
+ExpenseKanbanRenderer.template = 'hr_expense.KanbanRenderer';
+
+export class ExpenseDashboardKanbanRenderer extends ExpenseKanbanRenderer {}
+patch(ExpenseDashboardKanbanRenderer.prototype, 'expense_kanban_renderer_dashboard', ExpenseDashboardMixin);
```

Comment:
FYI there is an issue with patch that might not get fixed. If there ever is a patch on `KanbanController` or `KanbanRenderer` that extends `setup` it will not work with your views.
```suggestion
export class ExpenseKanbanController extends KanbanController {
    setup() {
        super.setup(...arguments);
    }
}
patch(ExpenseKanbanController.prototype, 'expense_kanban_controller_upload', ExpenseDocumentUpload);

export class ExpenseKanbanRenderer extends KanbanRenderer {
    setup() {
        super.setup(...arguments);
    }
}
patch(ExpenseKanbanRenderer.prototype, 'expense_kanban_renderer_qrcode', ExpenseMobileQRCode);
ExpenseKanbanRenderer.template = 'hr_expense.KanbanRenderer';

export class ExpenseDashboardKanbanRenderer extends ExpenseKanbanRenderer {}
patch(ExpenseDashboardKanbanRenderer.prototype, 'expense_kanban_renderer_dashboard', ExpenseDashboardMixin);
```